frontend/src/pages/ActivityPage.js

```diff
@@ -38,73 +29,118 @@ const fakeCommentData = [
-  const [activityData, setActivityData] = useState(fakeActivityData);
-  // this gets and sets loading
-  const [loading, setLoading] = useState({
-    running: false,
-    finished: false,
-    error: false
-  });
-  // this gets and sets the comment data
-  const [comments, setComments] = useState(fakeCommentData);
+class ActivityPage extends React.Component {
+  // Constructor
+  constructor(props) {
+    super(props);
+    this.state = {
+      activityData: null,
+      loadingData: {
+        finished: false,
+        error: false,
+      },
+      commentData: fakeCommentData,
+      id: props.id,
+    }
+  }
+
+  componentDidMount() {
+    fetch(`${api_url}/activity/${this.state.id}`, {
+      //mode: 'same-origin',
+    })
+      .then(res => {
+        console.log(res);
+        return res.json()
+      })
+      .then(
+        (result) => {
+          this.setState({
+            activityData: result.activity,
+            loadingData: {finished: true},
+          });
+        },
+        (error) => {
+          console.log(error);
+          this.setState({
+            loadingData: {
+              finished: false,
+              error
+            }
+          });
+        }
+      )
+  }
 
-  return (
-    <Container>
-      <Card>
-        <CardHeader>
-          {id}
-        </CardHeader>
-        <CardHeader
-          avatar={(
-            <Avatar
-              alt={`${activityData.author.username}'s profile picture`}
-              src={activityData.author.avatarURL}
-            >
-              {activityData.author.username}
-            </Avatar>
-          )}
-          title={activityData.author.fullname}
-          subheader={activityData.author.username}
-        >
-        </CardHeader>
-        <CardContent>
-          <Typography variant="h3" color="textSecondary">
-            Location
+  render() {
+    const {
+      activityData,
+      commentData,
+      loadingData,
+      id,
+    }=this.state;
+    if (loadingData.error) {
+      return <div> Error</div>;
+    }
+    else if(!loadingData.finished) {
+      return <CircularProgress/>;
+    }
+    else {
+      return (
+      <Container>
+        <Card>
+          <CardHeader
+            avatar={(
+              <Avatar
+                //these postedBy's need to be changed
+                alt={`${activityData.postedBy}'s profile picture`}
+                src={activityData.postedBy}
+              >
+                {activityData.postedBy}
+              </Avatar>
+            )}
+            title={activityData.postedBy}
+            subheader={activityData.postedBy}
+          >
+          </CardHeader>
+          <CardContent>
+            <Typography variant="h3" color="textSecondary">
+              {id}
           </Typography>
-          <Typography variant="body2" color="textSecondary">
-            Desc
+            <Typography variant="body2" color="textSecondary">
+              Desc
           </Typography>
```

Comment:
I forgot to mention this when we were going over it together, but we should change this to display the actual activity information instead of `{id}` and `Desc`.